src/command_modules/azure-cli-network/azure/cli/command_modules/network/dns_zone.py

```diff
@@ -22,3 +22,21 @@ def create_dns_zone(client, resource_group_name, zone_name, location='global', t
         kwargs['if_none_match'] = '*'
 
     return client.create_or_update(**kwargs)
+
+
+#pylint: disable=too-many-arguments
+def update_dns_zone(client, resource_group_name, zone_name, location='global', if_match=None,
+                    **operation_config):
+    body = {'location': location}
+
+    kwargs = {
+        'resource_group_name':resource_group_name,
+        'zone_name': zone_name,
+        'parameters': body,
+        'operation_config': operation_config
+    }
+
+    if if_match:
+        kwargs['if_match'] = if_match
+
+    return client.create_or_update(**kwargs)
```

Comment:
For an update, we simulate patch behavior, so using a PUT here (create_or_update) isn't consistent. Assuming you can update tags, if you wanted to update location, with this command, your tags would be lost.

Essentially you need to retrieve the zone object and conditionally overwrite the properties.